Modules/Microsoft365DSC/Modules/M365DSCUtil.psm1

```diff
@@ -3482,6 +3490,60 @@ function New-M365DSCMissingResourcesExample
     }
 }
 
+
+
+
+<#
+.Description
+This function validates there are no updates to the module or it's dependencies and no multiple versions are present on the local system.
+
+.Parameter Force
+Specifies that all dependencies should be forcefully imported again.
+
+.Example
+Test-M365DSCModuleValidity
+
+.Example
+Test-M365DSCModuleValidity -Force
+
+.Functionality
+Public
+#>
+function Test-M365DSCModuleValidity
+{
+    [CmdletBinding()]
+    param(
+        [parameter()]
+        [Switch]
+        $Force
+    )
+    $InformationPreference = 'Continue'
+
+    # validate only one installation of the module is present (and it's the latest version available from the psgallery)
+    $latestVersion = (Find-Module -Name 'Microsoft365DSC').Version
+    Install-Module -Name 'Microsoft365DSC' -Force -AllowClobber -RequiredVersion $latestVersion
+
+    [array]$modules = Get-Module -Name 'Microsoft365DSC' -List | Sort-Object Version -Descending
```

Comment:
Uninstall-M365DSCOutdatedDependencies already uninstalls previous versions of M365DSC. There is no need to implement it here.

Reply:
you're not wrong, but also not 100% correct.
the cmdlet does uninstall outdated versions of the m365dsc module - as long as theyre in the same psroot path youre executing the cmdlet from. Say you have multiple installations of powershell, e.g 7.1 and 5.2 in different locations, the following is gonna happen: [current module path](https://i.imgur.com/zklmZ0J.png), [All module paths](https://i.imgur.com/tfl5PD0.png)

Reply:
Fair point on that one. Then we should implement it within the uninstall-… function.